### DISCUSSION TITLE
Configure SAML client to not use line breaks in base64 elements

### DISCUSSION THREAD

**Discussion by mverbert** [keycloak/keycloak#14500] [Ideas]
Hi,
 after having upgraded from KC 17.0.1 to 19.0.2 (we missed 18.x series) we noticed that the base64 encoded elements now contain carriage return characters. I'm not sure that the change was intentional (at least I could not find it in the release notes, but I may have missed it), but it definitely breaks some clients (specifically urn:federation:MicrosoftOnline , but it may not be the only one). The only workaround solution we have so far is to rollback to the previous version version, but this is not a long term solution.

For instance we now have the following Modulus:
&lt;dsig:Modulus&gt;rIbMBqr8cc8smT4BUYPwppNOJ9/15eCMpJrf59hb1uUBW7f2aBVb8FzkxR/QcxSfU0oVdJOwfDXy&amp;#xD;
adB45IJWEHxHvMDn90942U2GPLsCTjhMXilTR26hsWCEjHxWaV8KAA+YiYbSkPK05piCUD9z4x5j&amp;#xD;
uIbAcOtQ3jLHxilFjHmgR9vn466S6wM3oH2N/2HwgSG282JD3eLY5NWAMQaP95Z2e3AdKM86Gh+i&amp;#xD;
lkPXbNcizS/Y8dkVatHnpVOC+gArDRar9BnY1tvVVuwhhVm3JHQ5ZbwmJcIpNnfakGgWuNt99quD&amp;#xD;
/dv7O0hDGGQ+Zef+yrRR3+SV6zwh5kvP0sXlew==&lt;/dsig:Modulus&gt;

and it used to be: 
&lt;dsig:Modulus&gt;rIbMBqr8cc8smT4BUYPwppNOJ9/15eCMpJrf59hb1uUBW7f2aBVb8FzkxR/QcxSfU0oVdJOwfDXyadB45IJWEHxHvMDn90942U2GPLsCTjhMXilTR26hsWCEjHxWaV8KAA+YiYbSkPK05piCUD9z4x5juIbAcOtQ3jLHxilFjHmgR9vn466S6wM3oH2N/2HwgSG282JD3eLY5NWAMQaP95Z2e3AdKM86Gh+ilkPXbNcizS/Y8dkVatHnpVOC+gArDRar9BnY1tvVVuwhhVm3JHQ5ZbwmJcIpNnfakGgWuNt99quD/dv7O0hDGGQ+Zef+yrRR3+SV6zwh5kvP0sXlew==&lt;/dsig:Modulus&gt;

I tentatively read through parts of RFC 2045 (https://www.ietf.org/rfc/rfc2045.txt) but I could not make up my mind if the new implementation is correct or not. If it is indeed correct, it would be a nice addition to be able to disable the line breaks at 76 characters for a particular client (if it's not, then it should be fixed as any regular bug).

Thanks for considering this feature request.

**Comment by bonnm** [comment]
I think this is a bug:
[https://github.com/keycloak/keycloak/issues/14529](https://github.com/keycloak/keycloak/issues/14529)
